lib/API/JSON/v1/Settings.php

```diff
@@ -81,6 +105,17 @@ public function set($settings = []) {
     }
   }
 
+  public function recalculateSubscribersScore() {
+    $this->statisticsOpensRepository->resetSubscribersScoreCalculation();
+    $task = new ScheduledTaskEntity();
+    $task->setType(SubscribersEngagementScore::TASK_TYPE);
+    $task->setScheduledAt(Carbon::createFromTimestamp($this->wp->currentTime('timestamp')));
+    $task->setStatus(ScheduledTaskEntity::STATUS_SCHEDULED);
+    $this->entityManager->persist($task);
```

Comment:
Is there a reason why you used the entityManager instead of the specific repository? I thought that we should use a repository when we can. I'm only curious.

Reply:
I didn't know you can do that. I thought you need to call `persist` on the entity manager. Can I use any repository or a specific one?

Reply:
> Can I use any repository or a specific one?
Yes, you can call persist and flush on the specific repository. I think that there is a check on the specific entity. You don't have to change it now. I was only curious.